dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -203,52 +188,41 @@ await foreach (ChatMessage message in choice.GetMessageStreaming(cancellationTok
     /// </summary>
     /// <param name="instructions">Optional chat instructions for the AI service</param>
     /// <returns>Chat object</returns>
-    private protected static ChatHistory InternalCreateNewChat(string? instructions = null)
+    private protected static OpenAIChatHistory InternalCreateNewChat(string? instructions = null)
     {
         return new OpenAIChatHistory(instructions);
     }
 
-    private protected List<ITextCompletionStreamingResult> InternalGetTextCompletionAsChat(string text, CompleteRequestSettings requestSettings, CancellationToken cancellationToken)
-    {
-        return new List<ITextCompletionStreamingResult>
-        {
-            new ChatCompletionAsTextResult(
-                (cancellationTokenInvoke) => this.InternalCompleteTextUsingChatStreamAsync(text, requestSettings, cancellationTokenInvoke),
-                (cancellationTokenInvoke) => this.InternalCompleteTextUsingChatAsync(text, requestSettings, cancellationTokenInvoke))
-        };
-    }
-
-    /// <summary>
-    /// Creates a completion for the prompt and settings using the chat endpoint
-    /// </summary>
-    /// <param name="text">The prompt to complete.</param>
-    /// <param name="requestSettings">Request settings for the completion API</param>
-    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
-    /// <returns>Text generated by the remote model</returns>
-    private async Task<string> InternalCompleteTextUsingChatAsync(
+    private protected async Task<IReadOnlyList<ITextCompletionResult>> InternalGetTextCompletionAsChatAsync(
         string text,
         CompleteRequestSettings requestSettings,
         CancellationToken cancellationToken = default)
     {
         ChatHistory chat = PrepareChatHistory(text, requestSettings, out ChatRequestSettings settings);
 
-        return await this.InternalGenerateChatMessageAsync(chat, settings, cancellationToken).ConfigureAwait(false);
+        return (await this.InternalGenerateChatMessageAsync(chat, settings, cancellationToken).ConfigureAwait(false))
+            .OfType<ITextCompletionResult>()
+            .ToList();
     }
 
-    private IAsyncEnumerable<string> InternalCompleteTextUsingChatStreamAsync(
+    private protected async IAsyncEnumerable<ITextCompletionStreamingResult> InternalGetTextCompletionStreamingAsChatAsync(
```

Comment:
Naming: I don't thing "AsChat" is quite right here, as that implies that we're casting something to a Chat, but it's actually doing the opposite (chat data "as" text completion result).

Suggestion: Instead of calling the model in these methods, could you just create a helper that takes in an IAsyncEnum and returns the other kind?
`IAsyncEnumerable<ITextCompletionStreamingResult> AsTextCompletionStreamingResult(
      IAsyncEnumerable<IChatCompletionStreamingResult> chatStreamResult
 )`
 
Not sure if this is practical, but it would mean there is only one actual implementation hitting Chat models, and one hitting the Completion models, and then the others are just casting/converting helper methods.